UI.cpp

```diff
@@ -0,0 +1,296 @@
+#include "UI.h"
+
+void UI::signInaccount(string name, string pass,int &id) {
+
+   Account account = this->service.signIn(name, pass, id);
+   Response message;
+   if(account.get_name()==name){
+       message.append("----------- Sign in successful -----------\n");
+       message.setSuccess(true);
+   }
+   else {
+       message.append("Username is taken\n");
+       message.setSuccess(false);
+   }
+    printMessage(message);
+    //return account;
+}
+
+Account UI::logInaccount(string name,string pass,Account account1){
+
+    Account account = this->service.login(name, pass, account1);
+    Response message;
+    if (account.get_name() != "") {
+        message.append("----------- Login successful -----------\n");
+        message.setSuccess(true);
+    }
+    else {
+        message.append("Login fail\n");
+        message.setSuccess(false);
+    }
+    printMessage(message);
+    return account;
+}
+
+
+std::string UI::user_name() {
+
+    std::cout << "ID: ";
+    std::string id;
+    std::cin >> id;
+    std::cout << std::endl;
+
+    return id;
+}
+
+std::string UI::password() {
+
+    std::cout << "Password: ";
+    char* s = new char[10];
+    int n = 0, x;
+    int i;
+    i = 0;
+    while (i < 18) {
+        x = _getch();
+        if (x != 13 and x != 8) {
+            s[n++] = x;
+            _putch('*');
+            i++;
+        }
+        else
+            if (x == 8 and n > 0)
+            {
+                s[n--];
+                _putch('\b');
+                _putch(' ');
+                _putch('\b');
+                i--;
+            }
+            else
+                if (x == 8 and i > 0)
+                    i--;
+                else
+                    if (x == 13)
+                        i = 18;
+
+    }
+
+    std::string str;
+    str.resize(n);
+    for (i = 0; i < n; i++)
+        str[i] = s[i];
+    return str;
+}
+
+void UI::printMessage(Response message) {
+
+    std::vector<std::string> messages = message.getAll();
+
+    for (int i = 0; i < int(messages.size()); i++) {
+        std::cout << std::endl << messages[i] << ' ';
+    }
+}
+Account UI::logIn(string name,string pass,Account account) {
+    return this->logInaccount(name,pass,account);
+}
+
+void UI::signIn(string name, string pass,int & id) {
+    this->signInaccount(name,pass,id);
+}
+
+Account UI::start_login() {
+    Account account;
+    int opt;
+    vector<Account>arr = service.display_users();
+    int id = stoi(arr[int(arr.size())-1].get_id());
+    string type;
+    std::cout << "----You need to loggin to have access----" << std::endl;
+    do {
+        std::cout << "  --Press 1 to have access to login" << std::endl << "  --Press 0 to sing in" << std::endl;
+        cin >> opt;
+        if (opt == 1) {
+            system("cls");
+            std::cout << std::endl << std::endl << "----------- Login -----------" << std::endl;
+            std::string name = user_name();
+            std::string pass = password();
```

Comment:
The name of these methods is confusing. What are they doing? They are called password and username. I should be able to see from the name that they get the username from the keyboard.